File: backend-laravel/app/Models/Company.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Support\Facades\Storage;

class Company extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'name',
        'industry',
        'description',
        'logo_path',
        'cover_image',
        'website',
        'location',
        'phone',
        'contact_email',
        'company_size',
        'founded_year',
        'verification_status',
        'admin_note',
        'dark_mode',
        'notify_messages',
        'notify_candidate_activity',
        'notify_quiz_results',
        'notify_post_activity',
    ];

    protected $casts = [
        'dark_mode' => 'boolean',
        'notify_messages' => 'boolean',
        'notify_candidate_activity' => 'boolean',
        'notify_quiz_results' => 'boolean',
        'notify_post_activity' => 'boolean',
    ];

    protected $appends = ['logo_url', 'cover_image_url'];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function jobPostings(): HasMany
    {
        return $this->hasMany(JobPosting::class);
    }

    public function getLogoUrlAttribute(): ?string
    {
        return $this->logo_path ? url(Storage::url($this->logo_path)) : null;
    }

    public function getCoverImageUrlAttribute(): ?string
    {
        return $this->cover_image ? url(Storage::url($this->cover_image)) : null;
    }
}

File: mobile-flutter/lib/models/company.dart
class Company {
  final int id;
  final int? userId;
  final String companyName;
  final String? industry;
  final String? location;
  final String? website;
  final String? companySize;
  final String? contactEmail;
  final String? description;
  final String? logo;

  Company({
    required this.id,
    this.userId,
    required this.companyName,
    this.industry,
    this.location,
    this.website,
    this.companySize,
    this.contactEmail,
    this.description,
    this.logo,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()) ?? 0,
      userId: json['user_id'] != null
          ? (json['user_id'] is int ? json['user_id'] : int.tryParse(json['user_id'].toString()))
          : null,
      companyName: json['company_name'] ?? json['name'] ?? 'Company',
      industry: json['industry'],
      location: json['location'],
      website: json['website'],
      companySize: json['company_size'],
      contactEmail: json['contact_email'],
      description: json['description'],
      logo: json['logo'] ?? json['logo_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'company_name': companyName,
      'industry': industry,
      'location': location,
      'website': website,
      'company_size': companySize,
      'contact_email': contactEmail,
      'description': description,
      'logo': logo,
    };
  }
}
